### DISCUSSION TITLE
Progress bar has a potential to introduce some best practices. Are they worth exploring?

### DISCUSSION THREAD

**Discussion by mbgower** [carbon-design-system/carbon#11637] [Ideas]
The Progress bar component is a maturing component. There are several considerations which could be emphasized by Carbon to improve the user experience, falling into 2 categories:
- accessibility improvements on state and outcome
- user experience improvement on persistence

**Is it worth exposing the Complete and Error icons programmatically?**
The component progresses to a resolution of either success or error. However, due to the component and poor cross-platform support, we cannot have faith that such information is exposed to the user agent or assistive technology. Is it worth giving the Success and Error icons an accessibility supported name and exposing them (they are currently aria-hidden)
<img width="45" alt="Screenshot 2022-06-16 at 10 11 22 AM" src="https://user-images.githubusercontent.com/14298245/174320154-5da423fa-47f1-42d7-a94a-6b399f021f89.png">
<img width="59" alt="Screenshot 2022-06-16 at 10 11 33 AM" src="https://user-images.githubusercontent.com/14298245/174320167-fae077f6-ab67-4559-af76-1fd9ad027371.png">

The advantages of this are:
1. The icons are _always_ used when the component resolves. So we have high confidence this information will be accessible and perceivable by everyone.
2. Rather than relying on a bunch of behind-the-scenes programmatic effort being maintained in each implementation, we just need to ensure developers don't actively hide the icon's accessible name. The odds of an exposed icon name remaining stable are much higher.
3. Where the author has not provided helper text on the outcome, the basic result would always be available in the accessible tree.

The disadvantage of this:
1. If there is accessible information supported by an AT, (i.e., `aria-valuenow`)it could potentially cause some redundant output. However, it does not appear that any of the supported aria attributes would do so. Some testing could confirm.
2. If the author has added helper text about the process being complete, it would add some redundancy to screen reader output.

**Is it worth making the progress bar's results persistent?**
Currently, the usage guidance allows teams the discretion to remove the entire progress bar and its results after it successfully completes. The main problem with this is that it's possible that someone does not see the success. If there is no persistent information about success, someone may assume it never ran.

This should only be a problem with success results; there is a requirement that items in error persist. I'd like Carbon to consider if at least the success icon could persist to show that an action has completed. It would take up minimal space. Especially if combined with the first idea (to expose the icon to the accessibility tree), it could be beneficial to a number of user groups.

**Comment by mbgower** [comment]
@laurenmrice Other considerations around Progress Bar from the review are captured as comments in the [Progress bar usage](https://ibm.box.com/s/jvu3n241qwsynpuoit2cb6hq2byx3p7g) and [style](https://ibm.box.com/s/xjt4n00xi71tijvaj0fsdbiw0gtau942) pdfs.

**Reply by laurenmrice** [reply]
@mbgower 

- I made an issue in the carbon-website repo to capture the content enhancements to the design documentation for progress bar. 
https://github.com/carbon-design-system/carbon-website/issues/3007

- As for making progress bar's results persistent, I think we could add this as a variant or prop so people can use this feature if they need to.

- I will leave it up to the carbon developers to decide on the other topics mentioned around exposing the Complete and Error icons programmatically.

**Reply by mbgower** [reply]
Thanks, @laurenmrice!  @joshblack I'll let your team think about the question around exposing the icon accessibly.
